core/storage/config/gae_models.py

```diff
@@ -120,6 +120,8 @@ class PlatformParameterModel(base_models.VersionedModel):
     rules = datastore_services.JsonProperty(repeated=True)
     rule_schema_version = (
         datastore_services.IntegerProperty(required=True, indexed=True))
+    is_feature_flag = (
```

Comment:
Wouldn't this require changing the constructors of currently-defined platform parameters?

Reply:
If you are talking about the parameters initialized here in the [platform_parameter_list.py](https://github.com/oppia/oppia/blob/develop/core/domain/platform_parameter_list.py) then I think we will not require that, reason being is that in the domain layer we already have a field `is_feature` which shows whether the parameter is a feature or not. When we initialize the feature-flags we do not have to pass the type of it as it is already of `bool` and for other platform parameter the value of `is_feature` is always false.

For the new platform-parameters that will introduce in future, I'm passing the `is_feature` field when creating datastore-entry.

Thanks

Reply:
I don't follow, sorry. Why are existing platform parameters and future platform parameters constructed according to different rules? We should have uniformity in how we do this, especially since the domain layer wouldn't know whether a platform parameter is "existing" or "new".

Reply:
Oh no, that is not the case.
So, what currently happens is when we create a platform parameter by initializing them in the [platform_parameter_list.py](https://github.com/oppia/oppia/blob/develop/core/domain/platform_parameter_list.py) file it does not directly get stored to the storage layer because in the storage layer we only store rules of platform-params, instead it gets stored to the domain layer variable - [parameter_registry](https://github.com/oppia/oppia/blob/develop/core/domain/platform_parameter_registry.py#L53) and as soon as we create a rule for the platform-param we create an entry for that in the storage layer.

What I was trying to convey in my previous comment was - For the domain layer we need not make any changes for the new field as we already have a `is_feature` field in the domain layer that keeps track of the same. Considering this, all of the existing platform-params have already a track of whether it is a feature or not in the domain layer and to update the storage layer, we are planning on doing that manually.
What I meant by the new platform-param was, when we create a new platform-param initially there will be no entry inside the storage layer and when we will create a new rule we make an entry to the storage layer, when doing that I'm passing value of `is_feature` to update the value of `used_as_feature_flag`.

Reply:
We will have a single entry in the storage layer. We have the field of type: List[platform_parameter_domain.PlatformParameterRuleDict]. The rules will just keep getting added to the list.

Yeah, I agree. We can avoid creating storage layer entries. I have removed the addition of the new field.